#ifndef MAINWINDOW_H
#define MAINWINDOW_H

#include <QMainWindow>
#include <QTcpSocket>
#include <QLabel>

QT_BEGIN_NAMESPACE
namespace Ui {
class MainWindow;
}
QT_END_NAMESPACE

class MainWindow : public QMainWindow
{
    Q_OBJECT

private:
    QTcpSocket *tcpClient; // socket
    QLabel *labSocketState; // 状态栏显示标签
    QString getLocalIP(); // 获取本机IP地址

public:
    MainWindow(QWidget *parent = nullptr);
    ~MainWindow();

private slots:
    void do_connected();
    void do_disconnected();
    void do_socketStateChange(QAbstractSocket::SocketState socketState);
    void do_socketReadyRead(); // 读取socket传入的数据
    void on_actConnect_triggered();

    void on_actDisconnect_triggered();

    void on_pushButton_clicked();

    void on_actClear_triggered();

private:
    Ui::MainWindow *ui;
};
#endif // MAINWINDOW_H
